### DISCUSSION TITLE
Description as Mandatory

### DISCUSSION THREAD

**Discussion by mbiagiotti** [kimai/kimai#3680] [Questions & Answers]
How can I set time tracking description mandatory (as customer, project and activity)?

**Comment by kevinpapst** [comment]
Out of the box you can't.

But you could use the [CustomContent plugin](https://www.kimai.org/store/custom-content-bundle.html) and add some javascript that sets the box as mandatory.

See https://www.kimai.org/documentation/plugin-custom-content.html

Snippet:

```javascript
document.addEventListener('show.bs.modal', (e) => {
    const desc = e.srcElement.querySelector('#timesheet_edit_form_description');
    if (desc !== null) {
        desc.required = true;
        const label = e.srcElement.querySelector('label[for=timesheet_edit_form_description]');
        if (label !== null && !label.classList.contains('required')) { label.classList.add('required'); }
    }
});
```

**Reply by mbiagiotti** [reply]
Thank you very much... I'll follow that way...

**Reply by EduNetEurope** [reply]
Hi Kevin,
is it possible to tweak the JavaScript code so that we can make the description box mandatory for certain users or teams only, or perhaps exclude specific users or teams from this requirement?
Considering that the Custom Field plugin has the requested functionality, wouldn't it be pretty simple to add the requirements and restrictions to the "description" field as well?
Thanks!

**Comment by Sarmingsteiner** [comment]
You could try https://gitlab.com/hmr-it/kimai2plugins/EmptyDescriptionCheckerBundle as a work-around.
